manifest/manifest.go

```diff
@@ -39,18 +39,15 @@ var (
 	}
 )
 
-type OnManifestChange func(ctx context.Context, prevManifest Manifest, errCh chan error)
+type OnManifestChange func(ctx context.Context, prevManifest Manifest) error
 
 type ManifestProvider interface {
-	// Run starts any background tasks required for the operation
-	// of the manifest provider.
-	Run(context.Context, chan error)
-	// Returns the list of gpbft options to be used for gpbft configuration
-	GpbftOptions() []gpbft.Option
-	// Set callback to trigger to apply new manifests from F3.
-	SetManifestChangeCallback(OnManifestChange)
-	// Manifest accessor
-	Manifest() Manifest
```

Comment:
Not particularly useful because this manifest will likely disagree with the manifest being used by F3 at the moment.